runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/DataflowRunnerTest.java

```diff
@@ -1151,6 +1167,49 @@ public void testStreamingWriteWithNoShardingReturnsNewTransformMaxWorkersUnset()
     testStreamingWriteOverride(options, StreamingShardedWriteFactory.DEFAULT_NUM_SHARDS);
   }
 
+
+  @Test
+  public void testMergingStatefulRejectedInStreaming() throws Exception {
```

Comment:
Same comments as in https://github.com/apache/beam/pull/3420 - potential deduplication, and reference JIRAs in source code next to creating the exceptions.